.github/workflows/push-main.yaml

```diff
@@ -0,0 +1,63 @@
+name: Push to Main
+
+on:
+  workflow_dispatch: {}
+  push:
+    branches:
+      - main
+
+jobs:
+  version-check:
+    runs-on: ubuntu-latest
+    outputs:
+      new_version: ${{ steps.check-version.outputs.new_version }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Check Version
+        id: check-version
+        run: |
+          version=$(grep -P -o 'version = "\d+\.\d+\.\d+"' pyproject.toml | grep -P -o '\d+\.\d+\.\d+')
+          tags=$(git tag -l '*.*.*')
+          match=$(echo ${tags} | grep -w -c ${version} | cat)
+          if [ "${match}" == "0" ]; then
+            echo "new_version=${version}" >> "${GITHUB_OUTPUT}"
+          else
+            echo "new_version=false" >> "${GITHUB_OUTPUT}"
+          fi
+
+  python-package:
+    runs-on: ubuntu-latest
+    needs: version-check
```

Comment:
- Isn't it an array? I.e. `needs: [version-check]`
- Do we also need the rock to succeed before releasing here?

Reply:
* Perhaps it *can* be an array but this works.
* No, this does not depend on the rock at all. This is just the pip package.